src/transformers/models/resnet/modeling_resnet.py

```diff
@@ -0,0 +1,435 @@
+        return pixel_values
+
+
+class ResNetStage(nn.Sequential):
+    """
+    A ResNet stage composed by stacked layers.
+
+    Args:
+        in_channels (`int`):
+            Number of input channels.
+        out_channels (`int`):
+            Number of output channels.
+        stride (`int`, *optional*, defaults to 1):
+            The first layer stride, used to downsample the input.
+        depth (`int`,*optional*, defaults to 2):
+            The number of layers.
+        layer_type (`str`, *optional*, defaults to `"basic"`):
+            The layer to use, it can be either `"basic"` (used for smaller models, like resnet-18 or resnet-34) or
+            `"bottleneck"` (used for larger models like resnet-50 and above).
+        activation (`int`, *optional*, defaults to `"relu"`):
+            The activation used by all layers.
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        stride: int = 2,
+        depth: int = 2,
+        layer_type: str = "basic",
+        activation: str = "relu",
+    ):
+        super().__init__()
+
+        layer = ResNetBottleNeckLayer if layer_type == "bottleneck" else ResNetBasicLayer
+
+        self.layers = nn.Sequential(
+            # downsampling is done in the first layer with stride of 2
+            layer(in_channels, out_channels, stride=stride, activation=activation),
+            *[layer(out_channels, out_channels, activation=activation) for _ in range(depth - 1)],
+        )
+
+
+class ResNetEncoder(nn.Module):
+    def __init__(self, config: ResNetConfig):
+        super().__init__()
+        self.stages = nn.ModuleList([])
+        # the first block doesn't downsample
+        self.stages.append(
+            ResNetStage(
+                config.hidden_sizes[0],
+                config.hidden_sizes[1],
+                stride=2 if config.downsample_in_first_stage else 1,
+                depth=config.depths[0],
+                layer_type=config.layer_type,
+                activation=config.hidden_act,
+            )
+        )
+        remaining_hidden_sizes = config.hidden_sizes[1:]
+        in_out_channels = zip(remaining_hidden_sizes, remaining_hidden_sizes[1:])
+        for (in_channels, out_channels), depth in zip(in_out_channels, config.depths[1:]):
+            self.stages.append(
+                ResNetStage(
+                    in_channels, out_channels, depth=depth, layer_type=config.layer_type, activation=config.hidden_act
+                )
+            )
+
+    def forward(
+        self, hidden_states: Tensor, output_hidden_states: bool = False, return_dict: bool = True
+    ) -> BaseModelOutputWithNoAttention:
+        all_hidden_states = () if output_hidden_states else None
+
+        for i, layer_module in enumerate(self.stages):
+            if output_hidden_states:
+                all_hidden_states = all_hidden_states + (hidden_states,)
+
+            hidden_states = layer_module(hidden_states)
```

Comment:
(nit) maybe more appropriate to call it `stage_module` (cause `layer_module` was adopted from BERT)

Reply:
indeed, or just `stage`